junit5/src/main/kotlin/io/dbkover/junit5/ExtensionContext.kt

```diff
@@ -23,7 +26,25 @@ internal fun ExtensionContext.findDataBaseConnectionFactory(): () -> Connection
             connectionMethod.invoke(requiredTestInstance) as Connection?
                 ?: throw RuntimeException("Connection not initialized correctly")
         }
+    } else if (connectionMethod == null) {
+        return { obtainConnectionFromProperties() ?: throw RuntimeException("Connection not initialized correctly") }
     }
 
     throw RuntimeException("Connection method is missing")
 }
+
+private fun obtainConnectionFromProperties(): Connection? {
+    val url: String? = System.getProperty("datasources.default.url")
+    val username: String? = System.getProperty("datasources.default.username")
+    val password: String? = System.getProperty("datasources.default.password")
+
+    if (url == null || username == null || password == null) {
+        return null
+    }
+
+    val props = Properties()
+    props.setProperty("user", username)
+    props.setProperty("password", password)
+
+    return DriverManager.getConnection(url, props)
```

Comment:
Since we are only using `user` and `password` in the properties we can also create the connection like below. What do you think about that? Or maybe you had another reason for going with the properties for this?

```suggestion
    return DriverManager.getConnection(url, username, password)
```

Reply:
I am indifferent about a or b, the reason I went with my suggestion was, if some consumer of this library should need to add additional props or simply not require username / password then I figured it was easier to append props rather than being bound to always require username and password. But it is an easy refactor if this should end up a requirement 🤷

Reply:
That could be an argument, but in the given code we enforce username / password anyway and have no support for adding additional properties.

I'm fine with both solutions (so you can resolve it if you prefer) - it's just shorter by specifying the username / password without the properties 😊

Reply:
I also think it is perfectly valid, that if someone would ever need it without username and password they should be able to create a PR for that 🤷